test/pylib/manager_client.py

```diff
@@ -248,6 +252,9 @@ async def rolling_restart(self, servers: List[ServerInfo], with_down: Optional[C
                 if idx2 != idx:
                     await self.server_not_sees_other_server(servers[idx2].ip_addr, s.ip_addr)
 
+            for s2 in servers_extra:
+                await self.server_not_sees_other_server(s2.ip_addr, s.ip_addr)
```

Comment:
Why not have a single loop over `servers_running` instead of two loops with `servers` and `servers_extra`?

Is it because of the `idx2 != idx` check? You could replace it with e.g. comparing server_ids instead

Reply:
Yes, it is because of this check ... I was actually wondering if we couldn't just compare the ServerInfo directly (but thought there might be some reason that it is done on indexes instead). Looking at the ServerInfo it looks like the comparison of servers (or even just server_ids as you mention) should work

Reply:
In whih case we wouldn't even need the other cleanup PR anyway as this would replace that part completely.

Reply:
Yes

Reply:
done